File: Pythonleetcode/Leetcode53.py
class Solution:

    def maxSubArray(self, nums):


        length = len(nums)

        for i in range(1, length):


            subMaxSum = max(nums[i] + nums[i - 1], nums[i])

            nums[i] = subMaxSum



        return max(nums)


print(Solution().maxSubArray([-2,1,-3,4,-1,2,1,-5,4]))

File: CppPractice/CppLeetcode/Leetcode53.cpp
#include <vector>
#include<algorithm>
using namespace std;

class Solution {
public:
    int maxSubArray(vector<int>& nums) 
    {
        
        if (nums.size()==1)
        {
            return nums[0];
            
        }
        int curMax = nums[0];
        int res = curMax;
        for (int i = 1; i < nums.size(); i++)
        {
            curMax = max(curMax+nums[i],nums[i]);
            res =max(res,curMax);

        }
        return res;
    }
};